header-rewrite-filter/header_processor.cc

```diff
@@ -224,17 +224,22 @@ namespace HeaderRewriteFilter {
 
             const Utility::MatchType match_type = Utility::StringToMatchType(*(start + 3));
 
-            std::string string_to_compare;
+            std::string string_to_compare = std::string(*(start + 4));
 
             switch (match_type) {
                 case Utility::MatchType::Exact:
-                    string_to_compare = std::string(*(start + 4));
-                    source_ = *(start + 1);
+                    source_ = std::string(*(start + 1));
                     matcher_ = [string_to_compare](std::string source) { return (source.compare(string_to_compare) == 0); };
                     break;
-                // TODO: implement this
+                case Utility::MatchType::Prefix:
+                    source_ = std::string(*(start + 1));
+                    matcher_ = [string_to_compare](std::string source) { return source.find(string_to_compare) == 0; };
```

Comment:
Nitpick: this is going to search the entire string, not just the prefix, which will be unnecessarily expensive if `source` is long. 

C++ std::string::find doesn't have a way to only search for a prefix, but you can either use `std::string::compare`:
```
string_to_compare.compare(0, source.size(), source, 0, source.size()) == 0 ) 
```
- note this is safe without checking `source.size() <= string_to_compare.size()`, `compare` is defined to search `If count1 > size() - pos1, the first substring is [pos1, size())`, see https://en.cppreference.com/w/cpp/string/basic_string/compare

Or invert the find and limit the search of `string_to_compare` to `source.size()`:
```
return string_to_compare.find(source.c_str(), 0, string_to_compare.size()) != std::string::npos;
```